### PR TITLE
Front page

### PR DESCRIPTION
**Pull request by shailee-m**
1. Added front-page screen
2. Add feeling and add Activity modals added
3. Common utils, services and some reusable components were also created in the process

### CODE REVIEW

**Comment by brylie** [comment]
The button text and icons skew to the right and, in some cases, clipped. 

Screenshot from 2020-09-11 12-18-40

There is a lot of spacing between the top navigation menu and the "Quick add" panel.

Screenshot from 2020-09-11 12-16-11

Please be attentive to these kinds of details.

**Review by brylie — CHANGES_REQUESTED** [review]
Please fix the button text and icon alignment. Please also fix the vertical spacing between the top of the page and the "quick add" panel.

**Comment by brylie** [comment]
It looks like you are using [Quasar Cards](https://quasar.dev/vue-components/card) instead of [Quasar buttons](https://quasar.dev/vue-components/button). Use the Quasar button for the buttons on the page.

```html
<q-btn color="green-9" push>
    <div class="row items-center no-wrap">
        <q-icon left name="fa-heartbeat" />

        <div class="text-center">
            Add<br>activity
        </div>
    </div>
</q-btn>
```

File: frontend/src/pages/Index.vue [hunk 1]
```diff
@@ -1,13 +1,15 @@
 <template>
   <q-page class="flex flex-center">
```
Comment (brylie) [inline frontend/src/pages/Index.vue]:
```suggestion
  <q-page class="flex q-mt-sm"">
```

Essentially, each of the Page templates would have the same `q-page` wrapper:

```html
<template>
  <q-page class="flex q-mt-sm">
    <front-page />
  </q-page>
</template>
```

Is there a way to share the `<q-page class="flex q-mt-sm">`across pages so they all behave the same without redundant code?

**Review by brylie — CHANGES_REQUESTED** [review]

File: frontend/src/pages/Index.vue (hunk 1 above)
Comment (shailee-m) [reply frontend/src/pages/Index.vue]: Done

**Comment by shailee-m** [comment]
> It looks like you are using [Quasar Cards](https://quasar.dev/vue-components/card) instead of [Quasar buttons](https://quasar.dev/vue-components/button). Use the Quasar button for the buttons on the page.
> 
> ```
> <q-btn color="green-9" push>
>     <div class="row items-center no-wrap">
>         <q-icon left name="fa-heartbeat" />
> 
>         <div class="text-center">
>             Add<br>activity
>         </div>
>     </div>
> </q-btn>
> ```

Buttons will not be good for that size, that's why used cards

**Comment by shailee-m** [comment]
> The button text and icons skew to the right and, in some cases, clipped.
> 
> Screenshot from 2020-09-11 12-18-40
> 
> There is a lot of spacing between the top navigation menu and the "Quick add" panel.
> 
> Screenshot from 2020-09-11 12-16-11
> 
> Please be attentive to these kinds of details.

Happening for very small screen size. Added ellipsis for the same.

**Comment by brylie** [comment]
> Buttons will not be good for that size, that's why used cards

We don't need the buttons to be the exact size as the current version. Padding within the button might make them larger if needed. 

In any case, we should use the q-btn element since these are buttons and should behave as such. Scroll down the page of the q-btn documentation and you will find examples of larger buttons containing icons.

**Comment by shailee-m** [comment]
> In any case, we should use the q-btn element since these are buttons and should behave as such. Scroll down the page of the q-btn documentation and you will find examples of larger buttons containing icons.

done. Also as it is a button, the icon and button text are on the same line. Separated by a small margin.

**Pull request merged**